WCAG-pagina: 2.4.4 Linkdoel (in context) (#890)
nl-design-system/documentatie: docs/wcag/2.4.4.mdx

# WCAG Succescriterium 2.4.4 Linkdoel (in context)

## W3C referenties

- Engelse tekst van het WCAG-succescriterium: [<span lang="en">2.4.4 Link Purpose (In Context)</span>](https://www.w3.org/TR/WCAG22/#link-purpose-in-context).
- Nederlandse vertaling van het WCAG-succescriterium: [2.4.4 Linkdoel (in context)](https://www.w3.org/Translations/WCAG21-nl//#linkdoel-in-context).
- Engelstalige informatie op <span lang="en">How to Meet WCAG</span>: [<span lang="en">Quick Reference 2.4.4 Link Purpose (In Context)</span>](https://www.w3.org/WAI/WCAG22/quickref/#link-purpose-in-context).
- Engelstalige toelichting: [<span lang="en">Understanding SC 2.4.4 Link Purpose (In Context)</span>](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html).

## Uitleg

De linktekst vertelt eenduidig aan de gebruiker waar de link naar toe gaat (het linkdoel).

Wat is een linktekst? Het tekst die zichtbaar is, maar ook de tekst die aan een gebruiker van hulpsoftware wordt voorgelezen.

Wat is het linkdoel? De locatie waar de link naartoe gaat. Bijvoorbeeld de contactpagina, een nieuwsbericht, een andere website of een download van een document.

> Wat betekent 'In context'? Het doel van de link kan ook door omringende content worden aangegeven. Dit moet dan wel ook voor gebruikers van hulpmiddelen duidelijk zijn, bijvoorbeeld door aanvullende tekst of een afbeelding naast de link. Dit laatste is onderwerp van veel discussie. Het beste is om altijd goed in de linktekst zelf te vertellen want het doel is.

Reviewer: nu met sierquotes:

```suggestion
Wat betekent ‘In context’? Het doel van de link kan ook door omringende content worden aangegeven. Dit moet dan wel ook voor gebruikers van hulpmiddelen duidelijk zijn, bijvoorbeeld door aanvullende tekst of een afbeelding naast de link. Dit laatste is onderwerp van veel discussie. Het beste is om altijd goed in de linktekst zelf te vertellen want het doel is.
```

Een eenduidige tekst is belangrijk voor alle gebruikers, zodat ze goed weten wat er gebeurt bij het aanklikken van een link. Screenreadergebruikers kunnen een lijst van alle links op een pagina oproepen en zo snel door de website navigeren, maar dan moeten de linkteksten wel onderscheidbaar zijn.

Het kan overduidelijk zijn waar een link heen gaat, maar soms is meer context nodig. Zijn er veel links met dezelfde linktekst op een pagina? Maak dan onderscheid tussen de linktekst door het toevoegen van meer informatie.

Voor ziende gebruikers kan duidelijk zijn waar een "Lees meer" link naar toe gaat, als deze linktekst onder een korte samenvatting staat. Om het voor screenreadergebruikers ook duidelijk te maken waar de link naar toe gaat kan er extra tekst worden toegevoegd die onzichtbaar, maar wel hoorbaar is.

Dit kan door gebruik te maken van CSS-classes zoals [.sr-only](https://tailwindcss.com/docs/screen-readers) of [.screen-reader-text](https://make.wordpress.org/accessibility/handbook/markup/the-css-class-screen-reader-text/).

In code:

```html
<a href="url">Lees meer<span class="sr-only"> over warmtepompen</a>.
```

> Of door het toevoegen van een aria-label` aan de link.

Reviewer: nu met extra ` 

```suggestion
Of door het toevoegen van een `aria-label` aan de link.

In code:

```html
<a href="url" aria-label="Lees meer over warmtepompen">Lees meer</a>.
```

Ook in tekstparagrafen is een duidelijke linktekst belangrijk. Vergelijk de scanbaarheid en duidelijkheid van de volgende twee voorbeelden.

- [Klik hier](https://nl.wikipedia.org/wiki/Waddenzee) om verder te lezen over de Waddenzee.
- Lees verder over [de Waddenzee](https://nl.wikipedia.org/wiki/Waddenzee).

> 'Klik hier' is de slechtste linktekst die er is. Alle gebruikers moeten eerst de hele zin lezen om te weten waar de link naar toe gaat.

Reviewer: ```suggestion
‘Klik hier’ is de slechtste linktekst die er is. Alle gebruikers moeten eerst de hele zin lezen om te weten waar de link naar toe gaat. Specifiek voor gebruikers met een screenreader die een lijst van links opvragen is het zeer onduidelijk als er één of meer keer ‘Klik hier’ in die lijst staat.
```

> Een goede tweede is het linken van een hele URL, zoals in 'Lees verder op [https://nl.wikipedia.org/wiki/Waddenzee](https://nl.wikipedia.org/wiki/Waddenzee)'.

Reviewer: 'goede tweede' verwarrend want het is juist niet goed :D iets hergeformuleerd voor extra duidelijkheid

```suggestion
Vermijd ook het linken van een hele URL, zoals in 'Lees verder op [https://nl.wikipedia.org/wiki/Waddenzee](https://nl.wikipedia.org/wiki/Waddenzee)'.
```

Goede tools voor het controleren voor dit succescriterium:

- De accessibility tree van je browser.
- Een screenreader, bijvoorbeeld [NVDA voor Windows](https://www.nvaccess.org/download/) of [VoiceOver/Safari](https://support.apple.com/nl-nl/guide/voiceover/welcome/mac) voor MacOS.

Controleer de linkteksten:

- Heeft de link een [toegankelijke naam](/wcag/4.1.2)?
- Beschrijft de linktekst duidelijk en eenduidig het linkdoel, of kan dit doel uit de omringende content worden begrepen?
- Is de linktekst en het linkdoel ook door gebruikers van hulpmiddelen te begrijpen?

## Veelgemaakte fouten

### Fout: De hele inhoud van een card/teaser/tile is de linktekst

Om een card/teaser/tile helemaal te linken wordt er een link om de hele inhoud gezet. Dit leidt tot een onleesbare linktekst voor screenreadergebruikers.

```html
<!-- Foute code, niet gebruiken -->
<a href="url">
  <img src="url" alt="Tulpen" />
  <h3>Wanneer bloeien tulpen?</h3>
  <p>
    Op welk moment bloeien tulpen precies en waar kun je de prachtigste tulpenvelden vinden? Ontdek hier alles over de
    lentepracht van Nederlandse bloemen <span>Lees meer ... </span>
  </p>
</a>
```

De linktekst wordt dan:

> 'Tulpen Wanneer bloeien tulpen? Op welk moment bloeien tulpen precies en waar kun je de prachtigste tulpenvelden vinden? Ontdek hier alles over de lentepracht van Nederlandse bloemen. Lees meer ...'

Reviewer: ‘Tulpen Wanneer bloeien tulpen? Op welk moment bloeien tulpen precies en waar kun je de prachtigste tulpenvelden vinden? Ontdek hier alles over de lentepracht van Nederlandse bloemen. Lees meer ...’

Deze constructie om een card te linken is een ramp om naar te luisteren met een screenreader.

Mogelijke oplossingen:

> Florian Schroiff beschrijft hoe een card aanklikbaar te maken via CSS met een [pseudo-element link overlay](https://www.frontendbbq.com/articles/accessible-teaser). In de Youtube video [What makes an accessible card?](https://www.youtube.com/watch?v=AH2NagMoYYE) onderzoekt Rian Rietveld de manieren om een card zo toegankelijk mogelijk te maken.

Reviewer: ```suggestion
Florian Schroiff beschrijft hoe een card aanklikbaar te maken via CSS met een [pseudo-element link overlay](https://www.frontendbbq.com/articles/accessible-teaser). In de YouTube video [What makes an accessible card?](https://www.youtube.com/watch?v=AH2NagMoYYE) onderzoekt Rian Rietveld de manieren om een card zo toegankelijk mogelijk te maken.
```

### Fout: Verkeerde linktekst voor een link met een afbeelding

Een link naar de voorpagina die een afbeelding van het logo bevat in de header is een bekend patroon.

> Een veel voorkomende fout is bijvoorbeeld de alt-tekst "Logo gemeente Voorbeeld".

Reviewer: ```suggestion
Een veel voorkomende fout is bijvoorbeeld de alt-tekst ‘Logo gemeente Voorbeeld’.
```

De alt-tekst beschrijft de afbeelding. Maar de alt tekst is hier ook de toegankelijke naam van de link.
In code:

```html
<!-- Dit is fout, niet gebruiken -->
<a href="link-url">
  <img src="img-url" alt="Logo gemeente Voorbeeld" />
</a>
```

> De toegankelijke naam van de link is in dit geval de alt-tekst en de linktekst wordt dus 'Logo gemeente Voorbeeld'.

Reviewer: ```suggestion
De toegankelijke naam van de link is in dit geval de alt-tekst en de linktekst wordt dus ‘Logo gemeente Voorbeeld’.
```

Dit is geen goede toegankelijke naam. De naam beschrijft niet waar de link naar toe gaat.

> Een goede oplossing is de alt-tekst te wijzigen in 'Gemeente Voorbeeld logo, naar de homepagina'.

Reviewer: ```suggestion
Een goede oplossing is de alt-tekst te wijzigen in ‘Gemeente Voorbeeld logo, naar de homepagina’.
```

```html
<a href="link-url">
  <img src="img-url" alt="Gemeente Voorbeeld logo, naar de homepagina" />
</a>
```

Dan vertel je wat er op de afbeelding staat en waar de link naar toe gaat.

> VoiceOver leest voor: 'Link, afbeelding, Gemeente Voorbeeld logo, naar de voorpagina'.

Reviewer: ```suggestion
VoiceOver leest dit voor als: ‘Link, afbeelding, Gemeente Voorbeeld logo, naar de voorpagina’.
```
